src/main/java/com/mewna/catnip/shard/manager/DefaultShardManager.java

```diff
@@ -173,74 +169,64 @@ private void loadShards() {
     private void startShard(final int id) {
         undeploy(id);
         catnip().logAdapter().info("Connecting shard {} (queue len {})", id, connectQueue.size());
+    
+        final var catnipShard = new CatnipShardImpl(catnip(), id, shardCount, catnip().initialPresence());
         
-        catnip().vertx().deployVerticle(new CatnipShard(catnip(), id, shardCount, catnip().initialPresence()), deployResult -> {
-            if(deployResult.failed()) {
-                catnip().logAdapter().error("Deploying shard {} failed, re-queueing!", id, deployResult.cause());
-                addToConnectQueue(id);
-                return;
-            }
-            shards.put(id, deployResult.result());
+        try {
+            catnipShard.start();
+            shards.put(id, catnipShard);
             catnip().logAdapter().info("Deployed shard {}(/{})", id, shardCount);
             connectShard(id);
-        });
+        } catch(final Exception e) {
+            catnip().logAdapter().error("Deploying shard {} failed, re-queueing!", id, e);
+            addToConnectQueue(id);
+        }
     }
     
     private void undeploy(final int id) {
-        final String deployment = shards.remove(id);
-        if(deployment != null) {
-            catnip().vertx().undeploy(deployment);
+        final CatnipShardImpl shard = shards.remove(id);
+        if(shard != null) {
+            shard.stop();
         }
     }
     
+    @SuppressWarnings("ResultOfMethodCallIgnored")
     private void connectShard(final int id) {
-        final String deploymentId = shards.get(id);
+        final CatnipShardImpl shard = shards.get(id);
         
-        if(deploymentId == null) {
+        if(shard == null) {
             catnip().logAdapter().error("Cannot find deployment ID of shard {}, re-queueing...", id);
```

Comment:
This shouldn't talk about deployment IDs anymore

Reply:
fixed in latest commit

Reply:
Log message too